src/mlpack/methods/hmm/hmm_impl.hpp

```diff
@@ -230,8 +237,8 @@ void HMM<Distribution>::Train(const std::vector<arma::mat>& dataSeq,
         << ")." << std::endl;
   }
 
-  initial.zeros();
-  transition.zeros();
+  arma::mat initial = arma::zeros(logInitial.n_elem);
+  arma::mat transition = arma::zeros(logTransition.n_rows, logTransition.n_cols);
```

Comment:
I think you could just estimate directly into `initialProxy` and `transitionProxy` here, instead of allocating `initial` and `transition`.

Reply:
I did not do that because once the proxy variables are going to removed from the code in later version, you don't need to edit the code, just need to remove the following from the code:
```
  initialProxy = initial;
  transitionProxy = transition;
```
and the rest remain the same. It's just for easy maintenance and hence it's in `train()` function, this extra copy it's not that a big deal from my point of view. If you disagree, it can sure be estimated directly.